core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2714,7 +2714,7 @@ object SparkContext extends Logging {
       case SparkMasterRegex.LOCAL_N_FAILURES_REGEX(threads, _) => convertToInt(threads)
       case "yarn" =>
         if (conf != null && conf.get(SUBMIT_DEPLOY_MODE) == "cluster") {
-          conf.getInt(DRIVER_CORES.key, 0)
+          conf.getInt(DRIVER_CORES.key, 1)
```

Comment:
I'm not sure this is correct. See the next 4 lines.

Reply:
really I think this should be turned into conf.get(DRIVER_CORES) which has the right default.

DRIVER_CORES = ConfigBuilder("spark.driver.cores")
    .doc("Number of cores to use for the driver process, only in cluster mode.")
    .intConf
    .createWithDefault(1)

Reply:
Hmm.

This is used eventually in the call to `NettyUtils.defaultNumThreads`. Changing this from 0 to 1 means that yarn-cluster apps will run with 1 thread for RPC handling instead of the default (which is `Math.min(availableCores, 8)`).

So I'm -1 on this change. Unless the problem above is handled in a different way.

Reply:
ah yes you are right, I missed that, agree